ocs_ci/ocs/utils.py

```diff
@@ -890,6 +893,48 @@ def collect_noobaa_db_dump(log_dir_path):
     )
 
 
+def save_pods_logs_to_file(
+    pod_patterns_containers, dir_name_prefix, patterns_to_log=None
+):
+    """
+    Save live logs of requested pods
+
+    Args:
+        pod_patterns_containers (dict): The patterns of the pods that
+            their logs should be saved as keys and the container names as values
+        dir_name_prefix (str): A suffix for the logs directory name. Should be the test case name
+        patterns_to_log (list): A pattern in the logs to look for and log
+
+    """
+    log_dir_path = os.path.join(
+        os.path.expanduser(ocsci_config.RUN["log_dir"]),
+        f"live_logs_{ocsci_config.RUN['run_id']}",
+    )
+    pods_containers = dict()
+    for pattern, containers in pod_patterns_containers.items():
+        pod_names = get_pod_name_by_pattern(pattern=pattern)
+        for pod_name in pod_names:
+            pods_containers.update({pod_name: containers})
+    live_logs_dir_path = os.path.join(
+        log_dir_path, f"{dir_name_prefix}_{uuid4().hex[:8]}"
+    )
+    create_directory_path(live_logs_dir_path)
+
+    if len(pods_containers) > 0:
+        for pod, container in pods_containers.items():
+            cmd = f"oc -n openshift-storage logs {pod} -c {container}"
+            logs_file = os.path.join(live_logs_dir_path, f"{pod}")
+            out = run_cmd(cmd)
+            if patterns_to_log:
+                for pattern_to_log in patterns_to_log:
+                    if pattern_to_log in out:
+                        log.info(
+                            f"The requested pattern {pattern_to_log} was found and it will be saved in {logs_file}"
+                        )
```

Comment:
Something is missing here... where is the save command in the file for every pattern? Or if it finds one occurrence of the `pattern_to_log`, it adds the logs to `log_file`? In that case, we can stop the for-loop after one occurrence and continue with the command:
`with open(logs_file, "w") as f:
                f.write(out)`

Reply:
the idea is not to save the pattern in a file. It's rather to inform, in the test log, if the pattern was found and in which location of log files it resides. The reason is that this new functionality saves the pod logs every 10 seconds, which results in having many sub-directories, sometimes in hundreds. So this indication is a hint where to look for in those files

Reply:
Ok, I understand.